core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -2420,6 +2420,44 @@ private[spark] object Utils extends Logging {
   }
 }
 
+private[spark] class CallerContext(
+   appName: Option[String] = None,
+   appID: Option[String] = None,
+   appAttemptID: Option[String] = None,
+   jobID: Option[Int] = None,
+   stageID: Option[Int] = None,
+   stageAttemptId: Option[Int] = None,
+   taskId: Option[Long] = None,
+   taskAttemptNumber: Option[Int] = None) extends Logging {
+
+   val AppName = if (appName.isDefined) s"_AppName_${appName.get}" else ""
```

Comment:
for the appName and the AppId, it feels like these should be obvious from the string so I'm not sure if we really need the extra _AppName_ and _AppID_ tags there.  Are there cases you don't think that is true?